### DISCUSSION TITLE
How to completely disable Firebase in unit tests?

### DISCUSSION THREAD

**Discussion by arietis** [firebase/firebase-ios-sdk#13812] [Q&A] [answered]
I have a test target that imports the @testable app module, and my app uses Firebase internally. I ensure Firebase is only configured when tests aren’t running, but I still notice some code from GoogleDataTransport being executed. This makes sense since some of its classes implement the +load class method. My question is: is there an effective way to completely prevent Firebase code from running during tests? In my case, it doesn’t serve any purpose during testing.

**Comment by paulb777** [comment, marked as answer]
I don't think there's a way. 

We'd like to migrate away from `+load` in the Firebase implementations in favor of on-demand loading, but haven't yet prioritized it. 

Feel free to create an issue if you'd like to track it.
